pkg/test/resourcefixture/testdata/basic/cloudbuild/v1alpha1/cloudbuildworkerpool/_http.log

```diff
@@ -499,7 +499,6 @@ X-Xss-Protection: 0
 
 GET https://cloudbuild.googleapis.com/v1/projects/${projectId}/locations/us-central1/workerPools/cloudbuildworkerpool-${uniqueId}?%24alt=json%3Benum-encoding%3Dint
 Content-Type: application/json
-x-goog-api-client: gl-go/1.22.4 gapic/1.16.1 gax/2.12.3 rest/UNKNOWN
```

Comment:
I hit this because my test env is `gl-go/1.23.0-20240419-RC02 gapic/1.16.1 gax/2.12.3 rest/UNKNOWN`

Reply:
We did have dropped this header.  But the problem is the `RemoveHeader` method only can drop the header in the capital-letters. This PR is to drop lower case like `x-goog-api-client` 
https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/tests/e2e/normalize.go#L270